web/tests/pm_list_data.test.js

```diff
@@ -162,6 +162,19 @@ test("get_conversations", ({override}) => {
     set_pm_with_filter("iago@zulip.com");
     pm_data = pm_list_data.get_conversations();
     assert.deepEqual(pm_data, expected_data);
+
+    override_rewire(people, "filter_people_by_search_terms", (users) => {
+        if (users.length === 1 && users[0].full_name === "Iago") {
+            return new Set(users);
+        }
+        return new Set();
+    });
+
+    pm_data = pm_list_data.get_conversations("search term does not matter for the test");
+    assert.deepEqual(
+        pm_data,
+        expected_data.filter((item) => item.recipients === "Iago"),
+    );
```

Comment:
Stubbing out logic functions for tests is not a recommended practice. The main point of our node tests is to verify logic; the stuff we want to stub out is UI/jquery code.

https://zulip.readthedocs.io/en/stable/testing/philosophy.html#integration-testing-or-unit-testing has some background, but I'd rather have a few simple end-to-end tests that the filtering does what's expected for some sample cases that we thought about in PR review -- and then if we change the algorithm, the changs to those tests in PR review is a natural mechanism to have reviewers for that future change be able to think about and discuss whether the adjustments to thos test cases are an improvement.

Reply:
fixed!